### PR TITLE
Make volume creation for nodes optional

### PR DESCRIPTION
**Pull request by lebauce**

### CODE REVIEW

File: volume_noop.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+description: >
+  A template which provides a volume for Docker storage
+
+parameters:
+  size:
+    description: >
+      size of a cinder volume in GB to allocate to docker for container/image storage
+    type: number
+
+  instance_uuid:
```
Comment (jprovaznik) [inline volume_noop.yaml]: this one should be probably removed

File: env_aop.yaml [hunk 2]
```diff
@@ -20,3 +20,4 @@ resource_registry:
   OOShift::LoadBalancer: loadbalancer_neutron.yaml
   OOShift::ContainerPort: sdn_openshift_sdn.yaml
   OOShift::IPFailover: ipfailover_keepalived.yaml
+  OOShift::DockerVolume: volume_docker.yaml
```
Comment (jprovaznik) [inline env_aop.yaml:23]: could you please add also mapping for DockerVolumeAttachment? + document this setting in README

File: fragments/node-boot.sh [hunk 3]
```diff
@@ -181,10 +181,13 @@ then
...
-docker_check_for_storage_device $DOCKER_VOLUME_DEVICE
+if [ -n "$VOLUME_ID" ]
+then
+    docker_check_for_storage_device $DOCKER_VOLUME_DEVICE
 
-# configure the external docker volume for LVM management
-docker_set_storage_device $DOCKER_VOLUME_DEVICE
+    # configure the external docker volume for LVM management
+    docker_set_storage_device $DOCKER_VOLUME_DEVICE
+fi
```
Comment (jprovaznik) [inline fragments/node-boot.sh:190]: Few lines bellow is duplicated:
/usr/bin/docker-storage-setup || notify_failure "Docker Storage setup failed"

I guess it's an old forgotten fragment - could you please remove it?

File: volume_noop.yaml (hunk 1 above)
Comment (lebauce) [inline volume_noop.yaml]: Done

File: fragments/node-boot.sh (hunk 3 above)
Comment (lebauce) [inline fragments/node-boot.sh:190]: Done

File: env_aop.yaml (hunk 2 above)
Comment (lebauce) [inline env_aop.yaml:23]: Done

**Pull request merged**